src/errors.rs

```diff
@@ -0,0 +1,47 @@
+use crate::AppInstruction;
+use backtrace::Backtrace;
+use std::panic::PanicInfo;
+use std::sync::mpsc::SyncSender;
+use std::{process, thread};
+
+pub fn handle_panic(info: &PanicInfo<'_>, send_app_instructions: &SyncSender<AppInstruction>) {
+    let backtrace = Backtrace::new();
+    let thread = thread::current();
+    let thread = thread.name().unwrap_or("unnamed");
+
+    let msg = match info.payload().downcast_ref::<&'static str>() {
+        Some(s) => *s,
+        None => match info.payload().downcast_ref::<String>() {
+            Some(s) => &**s,
+            None => "Box<Any>",
+        },
+    };
```

Comment:
I'm not sure I understand what's going on here. Can you help me out?

Reply:
(ping @kunalmohan because you said you don't always see these :) )

Reply:
`payload()` returns the payload attached with the panic, e.g. the string passed to `panic!()` or `.expect()`. It's return entity isn't statically typed but implements `Any` trait as mentioned [here](https://doc.rust-lang.org/std/panic/struct.PanicInfo.html#method.payload). Since it's mostly `&str` or `String` typed (as mentioned in docs), we check for them or return `"Box<Any>"` if it's not the case.

Reply:
I see... why is `Box<Any>` a static string ref though?

Reply:
I am not sure what you're asking. We'll have to return a single type (that implements the `Display` trait) from the different match arms, therefore the string ref (because the other arms return that). We can have any other message returned instead of `"Box<Any>"`.

Reply:
Yeah, that was the question. Thanks for the explanation. What do you think about having `msg` be an `Option`, and having an `and_then` instead of this match? Then we can handle the Some/None case in what is currently the `info.location()` match?

Something like:
```rust
    match ((info.location(), msg)) {
        (Some(location), Some(msg)) => { ... },
        (Some(location), None) => { ... },
        (None, Some(msg)) => { ... },
        (None, None) => { ... },
    }
```

Reply:
Sounds good.